06-classification_continued.Rmd

```diff
@@ -559,16 +559,257 @@ The overall workflow for performing K-nearest neighbour classification using `ti
 
 **Strengths:**
 
-1. Simple and easy to understand
-2. No assumptions about what the data must look like 
-3. Works easily for binary (two-class) and multi-class (> 2 classes) classification problems
+1. Simple, intuitive algorithm
+2. Few assumptions about what the data must look like 
+3. Works for binary (two-class) and multi-class (> 2 classes) classification problems
 
 **Weaknesses:**
 
-1. As data gets bigger and bigger, K-nearest neighbour gets slower and slower, quite quickly
-2. Does not perform well with a large number of predictors
+1. As the training data gets larger, K-nearest neighbours becomes very slow
+2. May not perform well with a large number of predictors
 3. Does not perform well when classes are imbalanced (when many more observations are in one of the classes compared to the others)
 
+## Predictor variable selection
+
+> This section is not required reading for the remainder of the textbook.
+
+Another potentially important part of tuning your classifier is to choose which
+variables from your data will be treated as predictor variables. Based
+on your initial exploratory analysis, you can choose
+anything from using a single predictor variable to using every variable in your
+data; the K-nearest neighbours algorithm accepts any number of
+predictors. However, it is **not** the case that using more predictors always
+yields better predictions! In fact, including irrelevant predictors can
+actually negatively affect classifier performance.
+
+Let's take a look at an example where K-nearest neighbours performs
+worse when given more predictors to work with. Here we have modified
+the breast cancer data to include the variables `Smoothness` and `Concavity`,
+as well as 500 irrelevant variables
```

Comment:
Does it have to be 500 to demonstrate this? That's a really exaggerated number... Also, since KNN suffers from the curse of dimensionality, is this poor prediction due to that, or due to irrelevant predictors?

Reply:
> curse of dimensionality, is this poor prediction due to that, or due to irrelevant predictors?

the curse of dimensionality in the context of K-NN is precisely the fact that irrelevant, noisy predictors that just add more or less random contributions to the pairwise distances. They're one and the same.

And yeah, we do need 500 (after looking at the plot you'll see why). But I will edit the writing to avoid explicitly stating that. I think it'll make it clearer to avoid mentioning it anyway.